src/shared/services/product.service.js

```diff
@@ -8,4 +8,45 @@ const getAllProducts = () => {
   return axios.get('/api/products');
 };
 
-export { getProductCategories, getAllProducts };
+const getProductCart = () => {
+  return axios.get('/api/user/cart');
+};
+
+const getProductWishList = () => {
+  const encodedToken = localStorage.getItem('userToken');
+  return axios.get('/api/user/wishlist', {
+    headers: {
+      authorization: encodedToken, // passing token as an authorization header
+    },
+  });
+};
+const saveProductInWishList = (product) => {
+  const encodedToken = localStorage.getItem('userToken');
+  return axios.post(
+    '/api/user/wishlist',
+    { product },
+    {
+      headers: {
+        authorization: encodedToken, // passing token as an authorization header
+      },
+    }
+  );
+};
+
+const deleteProductFromWishList = (productId) => {
+  const encodedToken = localStorage.getItem('userToken');
```

Comment:
You are repeating localStorage.getItem at many places, you can extract it out as a function.